docs/source/tutorials/fashion_mnist.ipynb

```diff
@@ -0,0 +1,1258 @@
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            embeddings = model.embeddings(images)\n",
+    "            embeddings_list.append(embeddings.cpu())\n",
+    "\n",
+    "    return torch.vstack(embeddings_list)\n",
+    "\n",
+    "\n",
+    "# Method for computing out-of-sample predicted probabilities\n",
+    "def compute_pred_probs(model, testloader):\n",
+    "    pred_probs_list = []\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            outputs = model(images)\n",
+    "            pred_probs_list.append(outputs.cpu())\n",
+    "\n",
+    "    return torch.vstack(pred_probs_list)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Create K folds of the dataset for cross validation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "K = 3  # Set to small value here to ensure quick runtimes, we recommend 5 or 10 in practice for more accurate estimates.\n",
+    "\n",
+    "kfold = KFold(n_splits=K, shuffle=True, random_state=0)\n",
+    "splits = kfold.split(transformed_dataset)\n",
+    "\n",
+    "train_id_list, test_id_list = [], []\n",
+    "\n",
+    "for fold, (train_ids, test_ids) in enumerate(splits):\n",
+    "    train_id_list.append(train_ids)\n",
+    "    test_id_list.append(test_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Compute out-of-sample predicted probabilities and feature embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_epochs = 3\n",
+    "patience = 2\n",
+    "batch_size = 4\n",
+    "\n",
+    "pred_probs_list, embeddings_list = [], []\n",
+    "\n",
+    "for i in range(K):\n",
+    "    print(f\"Training on fold: {i+1}\")\n",
+    "\n",
+    "    # Create train and test sets and corresponding dataloaders\n",
+    "    trainset = transformed_dataset.select(train_id_list[i])\n",
+    "    testset = transformed_dataset.select(test_id_list[i])\n",
+    "\n",
+    "    trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, num_workers=2)\n",
+    "    testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, num_workers=2)\n",
+    "\n",
+    "    # Train model\n",
+    "    model = train(trainloader, testloader, n_epochs, patience)\n",
+    "\n",
+    "    # Compute out-of-sample embeddings\n",
+    "    print(\"\\nComputing feature embeddings...\")\n",
+    "    fold_embeddings = compute_embeddings(model, testloader)\n",
+    "    embeddings_list.append(fold_embeddings)\n",
```

Comment:
@elisno Is the recommended practice? Vs. say just getting all embeddings from the model trained during first fold?

Didn't we find this leads to worse results because the embeddings from each fold's model are in a different global region of the embedding space?

Reply:
The notebook now uses a single `embedding_model` (which is just the model trained for the first fold).